File: Dagger/src/main/java/com/android/daggerdemo/Student.java
package com.android.daggerdemo;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class Student {
    private String name;
    private String age;
    private String grade;
    @Inject
    public Student() {
    }

    public Student(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "hashCode:" + hashCode() + "  Student{" +
                "name='" + name + '\'' +
                ", age='" + age + '\'' +
                ", grade='" + grade + '\'' +
                '}';
    }
}

File: Hilt/src/main/java/com/android/hilt/Student.kt
package com.android.hilt

import javax.inject.Inject

public class Student @Inject constructor() {

    var name: String? = ""
    var age: Int? = 0

    override fun toString(): String {
        return "HashCode:" + hashCode() + "   Student(name=$name, age=$age)"
    }

}